runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -119,6 +117,80 @@ public static String toReadableId(UUID id) {
         return Long.toHexString((id.getLeastSignificantBits()) & 0xFFFF);
     }
 
+
+    /**
+     * Verify that the stripes are not empty and that each segment has one stripe.
+     * @param segments segments to validate
+     */
+    private static void validateSegments(List<Layout.LayoutSegment> segments) {
+        checkNotNull(segments);
+        checkArgument(!segments.isEmpty());
+        long previousSegmentEndAddress = 0;
+        for (Layout.LayoutSegment segment : segments) {
+            checkState(segment.getStart() == previousSegmentEndAddress);
+            previousSegmentEndAddress = segment.getEnd();
+            // only supported for chain replication
+            checkArgument(segment.getReplicationMode() == Layout.ReplicationMode.CHAIN_REPLICATION);
+            // Since stripping is not supported, we can assume only one stripe exists
+            checkArgument(segment.getStripes().size() == 1);
+            // A stripe cannot be empty
+            checkArgument(!segment.getStripes().get(0).getLogServers().isEmpty());
+        }
+        // The last segment (i.e open segment) end address is -1 (denoting infinity)
+        checkState(previousSegmentEndAddress == -1);
+    }
+
+    /**
+     * This method selects a head node: the first node (in the open segment) that also exists in all previous
+     * segments.
+     * @param layout layout to use to find the head node
+     * @return head node
+     * @throws IllegalStateException if no such node exists
+     */
+    private static String selectHeadNode(Layout layout) {
+        checkNotNull(layout);
+        validateSegments(layout.getSegments());
```

Comment:
Segments and stripes can't be null https://github.com/CorfuDB/CorfuDB/blob/4ade684cd5f833bfffb26f5a84077b20d7020411/runtime/src/main/java/org/corfudb/runtime/view/Layout.java#L128

No need to check that they are not null one more time.

It's supposed by default that the layout is in a correct state. 
If you see any parameters in the layout that are not validates, it's better to add them into the Layout class

Reply:
Those are checks that happen in the constructor, but the Layout.java is not immutable. Might move these checks to the builder and make the Layout.java immutable in a subsequent patch.